test/saithrift/src/switch_sai_rpc_server.cpp

```diff
@@ -4214,6 +4216,119 @@ class switch_sai_rpcHandler : virtual public switch_sai_rpcIf {
+        sai_attribute_t sys_port_attr;
+        int max_sys_ports = 0;
+	sai_object_id_t sys_port_obj_id;
+
+        status = sai_api_query(SAI_API_SWITCH, (void **) &switch_api);
+        if (status != SAI_STATUS_SUCCESS) {
+            printf("sai_api_query failed!!!\n");
+            return SAI_NULL_OBJECT_ID;
+        }
+        attr.id = SAI_SWITCH_ATTR_TYPE;
+        status = switch_api->get_switch_attribute(gSwitchId, 1, &attr);
+        if (status != SAI_STATUS_SUCCESS) {
+            printf("get_switch_attribute failed!!!\n");
+            return SAI_NULL_OBJECT_ID;
+        }
+        if (attr.value.u32 != SAI_SWITCH_TYPE_VOQ) {
+            printf("Switch is not VOQ switch!!!\n");
+            return SAI_NULL_OBJECT_ID;
+        }
+        status = sai_api_query(SAI_API_SYSTEM_PORT, (void **) &sys_port_api);
+        if (status != SAI_STATUS_SUCCESS) {
+            printf("sai_api_query failed!!!\n");
+            return SAI_NULL_OBJECT_ID;
+        }
+
+        max_sys_port_attribute.id = SAI_SWITCH_ATTR_NUMBER_OF_SYSTEM_PORTS;
+        switch_api->get_switch_attribute(gSwitchId, 1, &max_sys_port_attribute);
+        max_sys_ports = max_sys_port_attribute.value.u32;
+
+        sys_port_list_object_attribute.id = SAI_SWITCH_ATTR_SYSTEM_PORT_LIST;
+        sys_port_list_object_attribute.value.objlist.list = (sai_object_id_t *) malloc(sizeof(sai_object_id_t) * max_sys_ports);
+        sys_port_list_object_attribute.value.objlist.count = max_sys_ports;
+        status = switch_api->get_switch_attribute(gSwitchId, 1, &sys_port_list_object_attribute);
+        if (status != SAI_STATUS_SUCCESS) {
+            printf("get_switch_attribute failed!!!\n");
+            free(sys_port_list_object_attribute.value.objlist.list);
+            return SAI_NULL_OBJECT_ID;
+        }
+
+        for (int i=0 ; i<max_sys_ports ; i++){
+           sys_port_attr.id = SAI_SYSTEM_PORT_ATTR_CONFIG_INFO;
+           status = sys_port_api->get_system_port_attribute(sys_port_list_object_attribute.value.objlist.list[i], 1, &sys_port_attr);
+           if (status != SAI_STATUS_SUCCESS) {
+               printf("get_system_port_attribute failed!!! for system_port 0x%lx\n", sys_port_list_object_attribute.value.objlist.list[i]);
+               continue;
+           }
+           if(sys_port_attr.value.sysportconfig.port_id == sys_port_id) {
+              sys_port_obj_id = sys_port_list_object_attribute.value.objlist.list[i];
+              free(sys_port_list_object_attribute.value.objlist.list);
+              return sys_port_obj_id;
+           }
+        }
+
+        printf("Didn't find system port port\n");
+        free(sys_port_list_object_attribute.value.objlist.list);
+        return SAI_NULL_OBJECT_ID;
+    }
+    void sai_thrift_get_system_port_attribute(sai_thrift_attribute_list_t& thrift_attr_list, const sai_thrift_object_id_t sys_port_oid) {
+      printf("sai_thrift_get_system_port_attribute for 0x%lx\n", sys_port_oid);
```

Comment:
Can we use SAI_THRIFT log instead of printf ?

Reply:
Fixed it